runtime/lua/vim/health/health.lua

```diff
@@ -493,11 +493,112 @@ local function check_external_tools()
+  end
+
+  local map = {
+    KITTY_WINDOW_ID = 'kitty',
+    ALACRITTY_SOCKET = 'alacritty',
+    ALACRITTY_LOG = 'alacritty',
+    WEZTERM_EXECUTABLE = 'wezterm',
+    KONSOLE_VERSION = function()
+      return 'konsole ' .. e.KONSOLE_VERSION
+    end,
+    VTE_VERSION = function()
+      return 'vte ' .. e.VTE_VERSION
+    end,
+  }
+
+  for key, val in pairs(map) do
+    local env = e[key] --- @type string?
+    if env then
+      return type(val) == 'function' and val() or val
+    end
+  end
+
+  return 'unknown'
+end
+
+local function check_sysinfo()
+  vim.health.start('Bug Report Information')
+
+  -- Use :version because vim.version().build returns "Homebrew" for brew installs.
+  local version_output = vim.api.nvim_exec2('version', { output = true }).output
+  local nvim_version = version_output:match('NVIM v[^\n]+') or 'unknown'
+  local commit_hash = nvim_version:match('%+g?(%x+)')
+  local version_for_report = nvim_version
+  if commit_hash then
+    version_for_report = nvim_version:gsub('%+g' .. commit_hash, ' neovim/neovim@' .. commit_hash)
+    local result = vim
+      .system({ 'git', 'ls-remote', 'https://github.com/neovim/neovim', 'HEAD' }, { text = true })
+      :wait()
+
+    if result.code == 0 then
+      local sha = result.stdout:match('^(%x+)')
+      local latest_commit = sha:sub(1, 10) --- @type string?
+      if latest_commit and commit_hash ~= latest_commit then
+        vim.health.warn(
+          string.format('Build is outdated. Local: %s, Latest: %s', commit_hash, latest_commit)
+        )
+      end
+    end
+  end
+
+  local os_info = vim.uv.os_uname()
+  local os_string = os_info.sysname .. ' ' .. os_info.release
+  local terminal = detect_terminal()
+  local term_env = vim.env.TERM or 'unknown'
+
+  vim.health.info('Nvim version: ' .. nvim_version)
+  vim.health.info('Operating system: ' .. os_string)
+  vim.health.info('Terminal: ' .. terminal)
+  vim.health.info('$TERM: ' .. term_env)
+
+  local body = vim.text.indent(
+    0,
+    string.format(
+      '## Problem:\n\n\n\n'
+        .. '## Steps to reproduce:\n\n```\nnvim --clean\n\n```\n\n'
+        .. '## Expected behavior:\n\n\n\n'
+        .. '## Nvim version (nvim -v):\n\n%s\n\n'
+        .. '## Vim (not Nvim) behaves the same?\n\n\n\n'
+        .. '## Operating system/version:\n\n%s\n\n'
+        .. '## Terminal name/version:\n\n%s\n\n'
+        .. '## $TERM environment variable:\n\n%s\n\n'
+        .. '## Installation:\n\n',
+      version_for_report,
+      os_string,
+      terminal,
+      term_env
+    )
+  )
+
+  local encoded_body = vim.uri_encode(body) --- @type string
+  local issue_url = 'https://github.com/neovim/neovim/issues/new?labels=bug&body=' .. encoded_body
+  vim.schedule(function()
+    local win = vim.api.nvim_get_current_win()
+    local buf = vim.api.nvim_win_get_buf(win)
+    if vim.bo[buf].filetype ~= 'checkhealth' then
+      return
+    end
+    _G._bugreport_open = function()
```

Comment:
As always, we want to "namespace" anything in a global scope.
```suggestion
    _G._nvim_health_bugreport_open = function()
```

Should probably do something like `_G._... = nil` on BufDelete.